### PR TITLE
Create functions for locale-sensitive date formatting

### PR DESCRIPTION
**Pull request by brandones**
## Requirements

- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).

## Summary

This creates date and time formatting functions which are locale sensitive and which limit the available formats to those recommended by the design team.

### CODE REVIEW

**Review by donaldkibet — APPROVED** [review]
LGTM Thanks @brandones 🚀

**Comment by brandones** [comment]
Thanks @donaldkibet !

CC @denniskigen @vasharma05 @samuelmale — I want to make sure this has support from the people who will be using it, developing UI features.

File: packages/framework/esm-utils/src/omrs-dates.ts [hunk 1]
```diff
@@ -79,36 +90,144 @@ export function toOmrsIsoString(date: DateInput, toUTC = false): string {
...
+ *
+ * This is created by concatenating the results of `formatDate`
+ * and `formatTime` with a comma and space. This agrees with the
+ * output of `Date.prototype.toLocaleString` for *most* locales.
+ */
+export function formatDatetime(date: Date, mode: FormatDateMode = "standard") {
+  const dateString = formatDate(date, mode);
+  const timeString = formatTime(date);
+  return `${dateString}, ${timeString}`;
+}
```
Comment (ibacher) [inline packages/framework/esm-utils/src/omrs-dates.ts:223]: From the designs, I notice that date and times are usually separated by `@`, e.g., `Today @ 11:21`, `30-Sep-2021 @ 18:42`. Shouldn't this then be the format we use here?

Comment (brandones) [reply packages/framework/esm-utils/src/omrs-dates.ts:223]: It's true. But see Ciarán's [comment](https://openmrs.slack.com/archives/CHP5QAE5R/p1639129372442800?thread_ts=1639110796.438300&cid=CHP5QAE5R) on Slack:

> I think we could reduce the date styles down to these two:
> dd-mmm-yyyy, hh:mm
> Today, hh:mm

**Pull request merged**